components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -1330,108 +1324,47 @@ private void updateLocalAndOutboundAuthenticationConfiguration(int applicationId
-            storeUseUserstoreDomainInLocalSubjectIdStmt.setInt(2, tenantID);
-            storeUseUserstoreDomainInLocalSubjectIdStmt.setInt(3, applicationId);
-            storeUseUserstoreDomainInLocalSubjectIdStmt.executeUpdate();
-        } finally {
-            IdentityApplicationManagementUtil.closeStatement(storeUseUserstoreDomainInLocalSubjectIdStmt);
-        }
-
-        PreparedStatement enableAuthzStmt = null;
-        try {
-            enableAuthzStmt = connection
-                    .prepareStatement(UPDATE_BASIC_APPINFO_WITH_ENABLE_AUTHORIZATION);
-            enableAuthzStmt.setString(1, localAndOutboundAuthConfig.isEnableAuthorization() ? "1" : "0");
-            enableAuthzStmt.setInt(2, tenantID);
-            enableAuthzStmt.setInt(3, applicationId);
-            enableAuthzStmt.executeUpdate();
-        } finally {
-            IdentityApplicationManagementUtil.closeStatement(enableAuthzStmt);
-        }
-
-        PreparedStatement storeSubjectClaimUri = null;
-        try {
-            storeSubjectClaimUri = connection
-                    .prepareStatement(UPDATE_BASIC_APPINFO_WITH_SUBJECT_CLAIM_URI);
-            // SUBJECT_CLAIM_URI=? WHERE TENANT_ID= ? AND ID = ?
-            storeSubjectClaimUri.setString(1, localAndOutboundAuthConfig.getSubjectClaimUri());
-            storeSubjectClaimUri.setInt(2, tenantID);
-            storeSubjectClaimUri.setInt(3, applicationId);
-            storeSubjectClaimUri.executeUpdate();
-        } finally {
-            IdentityApplicationManagementUtil.closeStatement(storeSubjectClaimUri);
-        }
+        PreparedStatement storeLocalAndOutboundConfigs = null;
 
         AuthenticationStep[] authSteps = localAndOutboundAuthConfig.getAuthenticationSteps();
 
-        if (authSteps == null || authSteps.length == 0) {
+        if (authSteps == null || authSteps.length == 0 || localAndOutboundAuthConfig.getAuthenticationType() == null) {
             // if no authentication steps defined - it should be the default behavior.
             localAndOutboundAuthConfig
                     .setAuthenticationType(ApplicationConstants.AUTH_TYPE_DEFAULT);
         }
 
         try {
-            if (localAndOutboundAuthConfig.getAuthenticationType() == null) {
-                // no authentication type defined - set to default.
-                localAndOutboundAuthConfig
-                        .setAuthenticationType(ApplicationConstants.AUTH_TYPE_DEFAULT);
-            }
-
-            updateAuthTypePrepStmt = connection
-                    .prepareStatement(UPDATE_BASIC_APPINFO_WITH_AUTH_TYPE);
-            // AUTH_TYPE=? WHERE TENANT_ID= ? AND ID = ?
-            updateAuthTypePrepStmt.setString(1, localAndOutboundAuthConfig.getAuthenticationType());
-            updateAuthTypePrepStmt.setInt(2, tenantID);
-            updateAuthTypePrepStmt.setInt(3, applicationId);
-            updateAuthTypePrepStmt.execute();
+            storeLocalAndOutboundConfigs = connection
```

Comment:
It would be good to declare this variable just about the try block?
ie. PreparedStatement storeLocalAndOutboundConfigs = null;

Also can't we use try-with-resources approach here?
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
Try-with-resource was not used because it has not been used elsewhere

Reply:
still needs to add the close statement if use try-with-resource. so, no need to use it

Reply:
Did get the explanation here?

Reply:
I mean try-with-resouce approach go against the way the `IdentityApplicationManagementUtil.closeStatement()` is implemented